### PR TITLE
fix: bug with fetching DSIS data from native

### LINKED ISSUE #25: fix issue with fetching data from DSIS
- [ ] try to connect using the Common Model for connecting to DSIS. It seems that we are trying fetch data and connecting using the Native Model.  The Common Model is what the DSIS team recommends to other apps when connecting to DSIS.
- - [ ] does the native model and common model give same response?
- [ ] suggestions from landmark support rpely
- - [ ] do not fetch all attributes. instead, use $select to pick the attributes we want
- - [ ]

### PR DESCRIPTION
**Pull request by KristianKjerstad**
Fix issue with fetching DSIS data.

- Utilize the $select query parameter to avoid fetching unnecessary data from DSIS, which reduce the response time drastically.
- Utilize the "odata.nextLink" attribute to fetch all data, instead of the $skip and $top parameters which seems to no longer be working

closes #25

### CODE REVIEW

**Review by tassab — CHANGES_REQUESTED** [review]
Nice implementation using the odata.nextLink and much faster than before with the select query parameter. 👍

Requests are repeated quite often due to empty response. Might waiting a few ms help? 🤷 Still much faster than before so not really an issue...

Order of attributes in CSV file has to be fixed. See comment.

File: src/use_cases/logs.py [hunk 1]
```diff
@@ -1,27 +1,21 @@
...
-                    dataframe[attribute] = ""
-            extract_attributes = dataframe.loc[:, attributes]
-            extract_attributes.to_csv(file, mode='a', index=False, header=False)
+    attribute_query_string = ",".join(
+        [attribute.replace(" ", "_") for attribute in attributes])
+    logs: pd.DataFrame = log_repository.get_logs_as_dataframe(attribute_query_string=attribute_query_string,
+                                                              project=project)
+    with open(output_file_path, 'w') as file:
+        file.write(",".join(attributes) + "\n")  # Write headers for CSV file.
+        logs.to_csv(file, mode='a', index=False, header=False)
```
[the remark this answers is not in the document]
Comment (KristianKjerstad) [reply src/use_cases/logs.py:22]: good catch!

**Review by tassab — APPROVED** [review]
Works fine now. I guess, the order of the columns isn't relevant?

**Comment by KristianKjerstad** [comment]
> Works fine now. I guess, the order of the columns isn't relevant?

I didn't give it much thought, but the order of the columns should probably be in the same order as the old training data so that we can expand the dataset without any issues.

Will create a fix for this!

**Review by eoaksnes — APPROVED** [review]
Nice

**Review by tassab — APPROVED** [review]
👍

**Pull request merged**